# Ujian Akhir Semester (4)

Nama   : Clarence Chandra Sentosa

NRP    : 1973010
    

# Training Classifier dengan Suppor Vector Machine (SVM)

# Load Breast Cancer Dataset

Pada kali ini saya akan menggunakan dataset yang sudah saya unduh terlebih dahulu pada website UCI Machine Learning.

In [23]:
# Code 1a
# Import Module Pandas

import pandas as pd

In [24]:
# Code 1b
# Load Dataset langsung dari UCI Machine Learning

df = pd.read_csv("https://archive.ics.uci.edu/ml/machine-learning-databases/breast-cancer-wisconsin/breast-cancer-wisconsin.data")

In [25]:
# Code 1c
# Menampilkan Data

df.head()

1000025  5   1  1.1  1.2  2 1.3  3  1.4  1.5  2.1
0  1002945  5   4    4    5  7  10  3    2    1    2
1  1015425  3   1    1    1  2   2  3    1    1    2
2  1016277  6   8    8    1  3   4  3    7    1    2
3  1017023  4   1    1    3  2   1  3    1    1    2
4  1017122  8  10   10    8  7  10  9    7    1    4

In [28]:
# Code 1d
# Menampilkan Dimensi Dataset

df.shape

(698, 11)

In [26]:
# Code 1e
# Menampilkan 5 Data Terakhir

df.tail()

1000025  5   1  1.1  1.2  2 1.3   3  1.4  1.5  2.1
693   776715  3   1    1    1  3   2   1    1    1    2
694   841769  2   1    1    1  2   1   1    1    1    2
695   888820  5  10   10    3  7   3   8   10    2    4
696   897471  4   8    6    4  3   4  10    6    1    4
697   897471  4   8    8    5  4   5  10    4    1    4

# Dataset Dimension

In [14]:
# Code 02

X, y = load_breast_cancer(return_X_y=True)
X.shape

(569, 30)

**Konsep berpikir code 02 :**

- Pertama-tama kita panggil method *load_breast_cancer* dengan menyertakan parameter (return_X_y=True).
- Lalu selanjutnya kita panggil *X.shape* dan kita coba run.

Bisa kita lihat pada output code 02 bahwa, jumlah baris pada **Breast Cancer** dataset tersebut berjumlah 569 baris dan 30 kolom.

# Training Set dan Testing Set

Kita akan membagi dataset tersebut kedalam training set dan testing set nya.

In [32]:
# Code 03

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,
                                                    y,
                                                    test_size=0.2,
                                                    random_state=2)

print(f'X_train shape {X_train.shape}')
print(f'X_test shape {X_test.shape}')
print(f'y_train shape {y_train.shape}')
print(f'y_test shape {y_test.shape}')

X_train shape (455, 30)
X_test shape (114, 30)
y_train shape (455,)
y_test shape (114,)


**Konsep berpikir code 03 :**
- Pertama-tama kita akan mencoba import terlebih dahulu modulnya dengan memanggil fungsi "from sklearn.model_selection import train_test_split".
- Selanjutnya kita akan memanggil "train_test_split" dengan menyertakan parameter "(X, y, test_size=0.2 (artinya 20% dari data kita akan dialokasikan untuk testing dataset, sedangkan 80% akan digunakan sebagai training dataset), random_state=2). Fungsi ini akan mengembalikan 4 buah kumpulan nilai yang perlu kita tangkap kedalam 4 variabel yaitu "X_train, X_test, y_train, dan y_test".
- Untuk selanjutnya, kita akan coba untuk menampilkan dimensi dari X_train dan X_test nya.

Bisa kita lihat pada code 03, untuk X_train jumlah datanya ada 455 dan untuk X_test ada 114 artinya X_train merupakan 80% dan X_test 20% dari keseluruhan dataset yang kita miliki.

# Classification dengan SVC (Support Vector Classifier)

In [37]:
# Code 04 (Import modul dan inisiasi)

from sklearn import svm

model = svm.SVC(kernel = 'linear',gamma = "auto",C = 0.3)

**Konsep berpikir code 04 :**
- Pertama-tama, import modul **SVM** pada *SkLearn* dengan memanggil *from sklearn import svm*.
- Lalu *svm.SVC* dengan menyertakan parameter *(kernel = 'linear',gamma = "auto",C = 0.3)* yang ditampung ke dalam variabel *model*.

**Penjelasan tentang *kernel*, *gamma*, dan *C***
- Metode **kernel** adalah kelas algoritma untuk analisis pola.
- Parameter **gamma** menentukan jarak pengaruh sampel data tunggal atau dikatakan menyesuaikan kelengkungan batas keputusan.
- **C** menentukan penalti atau berapa banyak sampel yang boleh ditempatkan di kelas yang berbeda.

In [39]:
# Code 05 (Fit Model)

model.fit(X_train, y_train)

SVC(C=0.3, gamma='auto', kernel='linear')

# Prediksi

Disini kita akan melakukan prediksi pada **Breast Cancer** dataset tersebut.

In [41]:
# Code 06 (Process Prediksi)

y_predicted = model.predict(X_test)

# Evaluasi 

Kita akan melakukan evaluasi **SVM** yang telah kita lakukan pada **Breast Cancer Dataset**

In [43]:
# Code 07 (Evaluasi nilai precision, recall, f1-score)

from sklearn.metrics import classification_report

print(classification_report(y_test, y_predicted))

              precision    recall  f1-score   support

           0       0.93      0.91      0.92        45
           1       0.94      0.96      0.95        69

    accuracy                           0.94       114
   macro avg       0.94      0.93      0.94       114
weighted avg       0.94      0.94      0.94       114



**Konsep berpikir code 07 :**
- Pertama kita akan mengimport modul terlebih dahulu dengan memanggigl method *from sklearn.metrics import classification_report*.
- Lalu kita tampilkan *(classification_report(y_test, y_predicted))* dengan print.

Pada output code 07 kita bisa lihat hasil evaluasi berupa nilai *precission*, *recall*, serta *f1-score* nya.

Demikianlah cara yang saya lakukan untuk melakukan training classifier dengan menggunakan *Support Vector Machine model* pada **Breast Cancer** dataset.

# "Banyak Belajar Biar Bisa Bantu Banyak Orang"

Written by : Clarence Code Pianist (1973010)